subsys/net/l2/openthread/openthread_utils.c

```diff
@@ -19,6 +19,56 @@
 
 #include "openthread_utils.h"
 
+int hex2bin(const char *aHex, uint8_t *aBin, uint16_t aBinLength)
+{
+    size_t      hexLength = strlen(aHex);
+    const char *hexEnd    = aHex + hexLength;
+    uint8_t *   cur       = aBin;
+    uint8_t     numChars  = hexLength & 1;
+    uint8_t     byte      = 0;
+
+    if ((hexLength + 1) / 2 > aBinLength)
+    {
+        return -1;
+    }
+
+    while (aHex < hexEnd)
+    {
+        if ('A' <= *aHex && *aHex <= 'F')
+        {
+            byte |= 10 + (*aHex - 'A');
+        }
+        else if ('a' <= *aHex && *aHex <= 'f')
```

Comment:
4. `else` should be placed in the same line as closing brace, so we are looking for something like `} else if (..) {`